### PR TITLE
feat: Slash Commands & Interactions [WIP?]

### PR DESCRIPTION
**Pull request by DjDeveloperr**
This PR implements Slash Commands & Interactions as of new API update, only what's documented though.
Reference: https://github.com/discord/discord-api-docs/pull/2295/

## Changes
- New gateway event `interactionCreate`
- Added payloads/types related to Interactions
- Added new structure `Interaction` with basic handling
- New `@slash` decorator

### CODE REVIEW

File: src/test/slash-cmd.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    {
+      name: 'only_smol',
+      description: 'Whether to show only baby animals',
+      type: 5,
+      required: false
+    }
+  ]
+}
+
+// fetch('https://discord.com/api/v8/applications/783937840752099332/commands', {
```
Comment (AkiaCode) [inline src/test/slash-cmd.ts:36]: Change fetch(... to `Url: https://discord.com/api/v8/applications/783937840752099332/commands`

Comment (DjDeveloperr) [reply src/test/slash-cmd.ts:36]: That I was testing, yet to introduce a `Client#slash` for a rich Slash Commands management!

Comment (AkiaCode) [reply src/test/slash-cmd.ts:36]: Oops, okay

**Comment by DjDeveloperr** [comment]
I think slash-support is complete, I can un-draft this and can be merged once update is rolled out completely?

File: src/models/slashClient.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        : APPLICATION_GUILD_COMMAND(
+            this.client.user?.id as string,
+            typeof guild === 'string' ? guild : guild.id,
+            id
+          ),
+      data
+    )
+    return this
+  }
+}
```
Comment (AkiaCode) [inline src/models/slashClient.ts:161]: Why is there no delete method?

Comment (DjDeveloperr) [reply src/models/slashClient.ts:161]: I forgot to push, hold on XD

Comment (AkiaCode) [reply src/models/slashClient.ts:161]: Okay

Comment (adolphin-hitler) [reply src/models/slashClient.ts:161]: >_>

Comment (adolphin-hitler) [reply src/models/slashClient.ts:161]: I’d say this is still a draft PR lmao.

Comment (DjDeveloperr) [reply src/models/slashClient.ts:161]: image

Isn't it already there

**Review by Helloyunho — APPROVED** [review]
Nice! Can't wait to use slash command

File: src/models/slashClient.ts (hunk 2 above)
Comment (AkiaCode) [reply src/models/slashClient.ts:162]: Oops

**Review by AkiaCode — APPROVED** [review]
lgtm

**Comment by Helloyunho** [comment]
image
Since the slash command is now in public beta, I'll merge this.

**Pull request merged**